googleapiclient/discovery.py

```diff
@@ -712,7 +707,7 @@ def method(self, **kwargs):
         raise TypeError('media_filename must be str or MediaUpload.')
 
       # Check the maxSize
-      if maxSize > 0 and media_upload.size() > maxSize:
+      if media_upload.size() and media_upload.size() > maxSize > 0:
```

Comment:
`media_upload.size()` isn't always defined (at least in one of the tests).  `None > 0`  will fail in python 3.